src/algebra/algebra/basic.lean

```diff
@@ -1430,18 +1459,27 @@ instance [I : add_comm_group M] : add_comm_group (restrict_scalars R A M) := I
 instance restrict_scalars.module_orig [semiring A] [add_comm_monoid M] [I : module A M] :
   module A (restrict_scalars R A M) := I
 
-variables [comm_semiring R] [semiring A] [algebra R A]
-variables [add_comm_monoid M] [module A M]
+variables [semiring A] [add_comm_monoid M] [module A M]
 
-/--
-When `M` is a module over a ring `A`, and `A` is an algebra over `R`, then `M` inherits a
-module structure over `R`.
-
-The preferred way of setting this up is `[module R M] [module A M] [is_scalar_tower R A M]`.
--/
-instance : module R (restrict_scalars R A M) :=
+/-- When `M` is a module over a semiring `A`, and `A` is an algebra over `R`, then `M` inherits a
+module structure over `R`. -/
+instance [comm_semiring R] [algebra R A] : module R (restrict_scalars R A M) :=
 module.comp_hom M (algebra_map R A)
 
+namespace restrict_scalars
+
+/-- `restrict_scalars` is an equivalence of modules over the semiring `A`. -/
+def linear_equiv : restrict_scalars R A M ≃ₗ[A] M := linear_equiv.refl A M
```

Comment:
I thought about adding this in the past, but concluded there was little point because it's just a convoluted way to spell `id`. Do you actually end up needing this?

Reply:
I don't strictly need it but I claim that the way I use it here https://github.com/leanprover-community/mathlib/pull/7808/files#diff-8a09a95c8c28517dfa1a667d83cfdc6dff8cfbe366abb3c0e6000dc87bea4176R150 justifies its existence.